[CDCSDK] Connector deployment fails when using yb_grpc replication slot plugin

Jira Link: [DB-22356](https://yugabyte.atlassian.net/browse/DB-22356)
### Description

I ran the existing gRPC stream creation test after creating the replication slot using the yb_grpc plugin instead of yboutput.
The connector deployment fails during stream creation with the following error:
```
AssertionError: Failed to add connector stream.
res.reason="Internal Server Error"
res.content=b'{"error_code":500,"message":"No enum constant io.debezium.connector.yugabytedb.YugabyteDBConnectorConfig.LogicalDecoder.YB_GRPC"}'
```

### Source connector version

quay.io/yugabyte/debezium-connector:dz.1.9.5.yb.grpc.2026.1

### Connector configuration

2026-07-13 20:57:26,283:DEBUG: add connector connector_name='ybconnector_cdc_f337bb_test_cdc_1dad11_hash' stream_id='rs_cdc_f337bb_bb5eab' db_name='cdc_f337bb' connector_host='172.151.29.127' table_list=['test_cdc_1dad11_hash'] {'name': 'ybconnector_cdc_f337bb_test_cdc_1dad11_hash', 'config': {'database.master.addresses': '172.151.28.197:7100,172.151.30.69:7100,172.151.21.96:7100', 'database.hostname': '172.151.28.197:5433,172.151.30.69:5433,172.151.21.96:5433', 'database.port': 5433, 'database.masterhost': '172.151.30.69', 'database.masterport': '7100', 'database.user': 'yugabyte', 'database.password': 'yugabyte', 'database.dbname': 'cdc_f337bb', 'snapshot.mode': 'initial', 'admin.operation.timeout.ms': 600000, 'socket.read.timeout.ms': 300000, 'max.connector.retries': '10', 'operation.timeout.ms': 600000, 'topic.creation.default.compression.type': 'lz4', 'topic.creation.default.cleanup.policy': 'delete', 'topic.creation.default.partitions': 2, 'topic.creation.default.replication.factor': '1', 'tasks.max': '5', 'connector.class': 'io.debezium.connector.yugabytedb.YugabyteDBgRPCConnector', 'database.server.name': 'ybconnector_cdc_f337bb_test_cdc_1dad11_hash', 'plugin.name': 'yb_grpc', 'slot.name': 'rs_cdc_f337bb_bb5eab', 'publication.autocreate.mode': 'disabled', 'publication.name': 'pub_cdc_f337bb_test_cdc_1dad11_hash', 'table.include.list': 'public.test_cdc_1dad11_hash', 'database.sslrootcert': '/kafka/ca.crt'}}
2026-07-13 20:57:26,283: INFO: Listing all connectors

### YugabyteDB version

2.31.0.0-b999878

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [x] I confirm this issue does not contain any sensitive information.

[DB-22356]: https://yugabyte.atlassian.net/browse/DB-22356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

--- a/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java
+++ b/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java
@@ -403,6 +403,23 @@ public class YugabyteDBConnectorConfig extends RelationalDatabaseConnectorConfig
             public boolean supportsTruncate() {
                 return false;
             }
+        },
+        // Plugin name for gRPC CDC streams created via the PostgreSQL replication-slot syntax.
+        YB_GRPC("yb_grpc") {
+            @Override
+            public MessageDecoder messageDecoder(MessageDecoderContext config) {
+                return new YbProtoMessageDecoder();
+            }
+
+            @Override
+            public String getPostgresPluginName() {
+                return getValue();
+            }
+
+            @Override
+            public boolean supportsTruncate() {
+                return false;
+            }
         };
 
         private final String decoderName;
@@ -773,7 +790,7 @@ public class YugabyteDBConnectorConfig extends RelationalDatabaseConnectorConfig
             .withDefault(DEFAULT_PLUGIN_NAME)
             .withDescription("The name of the Postgres logical decoding plugin installed on the server. " +
                     "Supported values are '" + LogicalDecoder.YBOUTPUT.getValue()
-                    + "'. " +
+                    + "' and '" + LogicalDecoder.YB_GRPC.getValue() + "'. " +
                     "Defaults to '" + LogicalDecoder.YBOUTPUT.getValue() + "'.");
 
     public static final Field SLOT_NAME = Field.create("slot.name")
@@ -786,7 +803,7 @@ public class YugabyteDBConnectorConfig extends RelationalDatabaseConnectorConfig
             .withValidation(YugabyteDBConnectorConfig::validateReplicationSlotName)
             .withDescription("The name of the YSQL logical decoding slot created for streaming changes from a plugin." +
                     "Defaults to 'debezium");
-    
+
     public static final Field DROP_SLOT_ON_STOP = Field.create("slot.drop.on.stop")
             .withDisplayName("Drop slot on stop")
             .withType(Type.BOOLEAN)
@@ -1913,7 +1930,7 @@ public class YugabyteDBConnectorConfig extends RelationalDatabaseConnectorConfig
                 LOGGER.warn(
                         String.format("Error while trying to %s filtered publication. Will retry, attempt %d out of %d",
                                 isUpdate ? "update" : "create", retryCount, maxAttempts));
-                    
+
                 pauseBetweenRetries(config.getLong(RETRY_DELAY_MS));
             }
         }
@@ -1931,7 +1948,7 @@ public class YugabyteDBConnectorConfig extends RelationalDatabaseConnectorConfig
                 LOGGER.trace("Ignoring table {} as it's not included in the filter configuration", tableId);
                 continue;
             }
-            
+
             LOGGER.trace("Adding table {} to the list of captured tables", tableId);
             capturedTables.add(tableId);
         }

--- a/src/test/java/io/debezium/connector/yugabytedb/YugabyteDBPublicationReplicationTest.java
+++ b/src/test/java/io/debezium/connector/yugabytedb/YugabyteDBPublicationReplicationTest.java
@@ -36,7 +36,7 @@ import static org.junit.jupiter.api.Assertions.*;
 /**
  * Unit tests to verify that connector works well with Publication and Replication slots.
  * The minimum service version required for these tests to work is 2.20.2
- * 
+ *
  * @author Sumukh Phalgaonkar (sumukh.phalgaonkar@yugabyte.com)
  */
 public class YugabyteDBPublicationReplicationTest extends YugabyteDBContainerTestBase {
@@ -92,7 +92,7 @@ public class YugabyteDBPublicationReplicationTest extends YugabyteDBContainerTes
         TestHelper.execute(String.format(TestHelper.createPublicationForTableStatement, "pub", "t1"));
         TestHelper.execute(TestHelper.createReplicationSlotStatement);
 
-        Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot"); 
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot");
         startEngine(configBuilder);
         final long recordsCount = 1;
 
@@ -107,6 +107,26 @@ public class YugabyteDBPublicationReplicationTest extends YugabyteDBContainerTes
     }
 
     @Test
+    public void testYbGrpcPluginReplicationStreamingConsumption() throws Exception {
+      TestHelper.execute(String.format(TestHelper.createPublicationForTableStatement, PUB_NAME, "t1"));
+      TestHelper.execute("SELECT pg_create_logical_replication_slot('" + SLOT_NAME + "', 'yb_grpc');");
+
+      Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", PUB_NAME, SLOT_NAME);
+      configBuilder.with(YugabyteDBConnectorConfig.PLUGIN_NAME, YugabyteDBConnectorConfig.LogicalDecoder.YB_GRPC.getValue());
+      startEngine(configBuilder);
+      final long recordsCount = 1;
+
+      awaitUntilConnectorIsReady();
+
+      // Introducing sleep to ensure that we insert the records after bootstrap is completed.
+      Thread.sleep(30000);
+
+      insertRecords(recordsCount);
+
+      verifyPrimaryKeyOnly(recordsCount);
+    }
+
+    @Test
     public void testPublicationReplicationSnapshotConsumption() throws Exception {
         TestHelper.execute("CREATE TABLE IF NOT EXISTS t2_snapshot (id int primary key);");
         TestHelper.execute(String.format(TestHelper.createPublicationForTableStatement, "pub", "t2_snapshot"));
@@ -116,7 +136,8 @@ public class YugabyteDBPublicationReplicationTest extends YugabyteDBContainerTes
         final int recordsCount = 1000;
         TestHelper.executeBulk(insertStatement, recordsCount);
 
-        Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot"); 
+        Configuration.Builder configBuilder =
+            TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot");
         configBuilder.with(YugabyteDBConnectorConfig.SNAPSHOT_MODE, YugabyteDBConnectorConfig.SnapshotMode.INITIAL.getValue());
         startEngine(configBuilder);
 
@@ -566,8 +587,6 @@ public class YugabyteDBPublicationReplicationTest extends YugabyteDBContainerTes
         }).get();
     }
 
-    
-
     private void verifyPrimaryKeyOnly(long recordsCount) {
         List<SourceRecord> records = new ArrayList<>();
         waitAndFailIfCannotConsume(records, recordsCount);
@@ -577,7 +596,7 @@ public class YugabyteDBPublicationReplicationTest extends YugabyteDBContainerTes
             assertValueField(records.get(i), "after/id/value", i);
         }
     }
-    
+
     private boolean dropReplicationSlot() throws Exception {
         try (YugabyteDBConnection ybConnection = TestHelper.create();
              Connection connection = ybConnection.connection()) {

